externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSQLSessionManager.scala

```diff
@@ -69,6 +75,18 @@ class SparkSQLSessionManager private (name: String, spark: SparkSession)
       setSession(handle, sessionImpl)
       info(s"$user's session with $handle is opened, current opening sessions" +
       s" $getOpenSessionCount")
+
+      if (singleSparkSessionInitialized.compareAndSet(false, true)
```

Comment:
1. We need to move this up to L74 or higher.

2. nit:
```scala
val singleSparkSession = this.conf.get(ENGINE_SINGLE_SPARK_SESSION)
```
3. we can execute `ENGINE_SESSION_INITIALIZE_SQL` once when the engine startup instead of hacking here

Reply:
Addressed 1 and 2.
Have some thoughts on 3, one difference between single session mode and isolate session mode is that the former uses the engine startup session but the latter does not. Is it necessary also to execute `ENGINE_SESSION_INITIALIZE_SQL` following engine startup for the engine startup session?

Reply:
I think it's ok, as the main purpose of init sqls is always to bring up some external stuff alive or visible